pkg/auth/oauth2authenticator.go

```diff
@@ -245,6 +247,13 @@ func (o *oAuth2Authenticator) IsSupported() bool {
 }
 
 func (o *oAuth2Authenticator) persistToken(token *oauth2.Token) error {
+	if audience, err := readAudience(token); err == nil && audience != "" {
```

Comment:
Suggestion: how about making this part of the default function for the API URL instead of adapting the `oAuth2Authenticator` and persisting the api url. To me this logic seems to be more snyk specific then an oauth specific change.

Reply:
I think I see what you mean... parse it out of the stored oauth token if it's configured? Yes... that's a much better approach!

Reply:
Updated, with a unit test